### DISCUSSION TITLE
Storybook as documentation tool for Angular

### DISCUSSION THREAD

**Discussion by ManuelEssmeister** [storybookjs/storybook#21012] [Help]
Hi,
is it possible (and adviced) to use Storybook as a full documetation tool for Angular applications and libraries? I've found a lot of documentation about creating stories for components but an Angular app/library consists of more than just components. I'm talking about services, pipes, directives etc.

I managed to setup documan which generates a documentation.json file with all the in-code documentation of all my components, services. etc. I also managed to use it for component documentation generation in stories. So all the inputs use my in-code documentation. But how can i get storybook to display the services, pipes, etc.? Is there a way to create a story for a service that automatically uses the in-code documentation that is already present during code-completion?

Any advice is very much appreciated.

ps.: I am using the latest beta of Storybook 7 with an NX workspace and Angular 15.

**Comment by bzh2610** [comment]
Hi,

With Angular, Storybook depends on [compodoc](https://github.com/compodoc/compodoc#readme) to generate the documentation.json file which is used to reference your stories and components (if you have the option "compodoc" set to true inside of storybook start script within angular.json).

It's a quite powerful tool for indexing and documenting your components, services, pipes and directives. 
Storybook uses compodoc to generate a .json file but you can also use it as a web server :

To run compodoc a server, you can add the following command to your package.json scripts :
`"compodoc:serve": "compodoc -s",`

Here's a gif of the tool's UI from compodoc :
![Compodoc gif](https://raw.githubusercontent.com/compodoc/compodoc/master/screenshots/main-view.gif)

I hope this helps
